mteb/evaluation/MTEB.py

```diff
@@ -317,6 +317,17 @@ def run(
                     if verbosity >= 1:
                         logger.info(f"Scores: {results}")
 
+                # if the model is a reranker, get the first stage model info
+                if "first_stage" in kwargs:
```

Comment:
It's really crucial to store what the first stage model is, but it ended up being really hard to get that information from a random model. Perhaps if there was some designated name in the DRESModel or something, but otherwise I can't pull anything out of SentenceTranformer models that say the name consistently.

Reply:
Based on requests from @KennethEnevoldsen, from SentenceTransformers 3.0 onwards it will be possible to get the model name and revision of a model like so:
```python
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("all-mpnet-base-v2")
print(model.model_card_data.base_model)
print(model.model_card_data.base_model_revision)
```

(This data is also used in the upcoming automatic model card generation, hence why it's stored under `model_card_data` rather than e.g. `info`)

- Tom Aarsen

Reply:
This looks really cool! Seeing that this is in progress, I'll leave it be for now and we can update it in the future.  Seems like a great idea! 

And thanks for adding this to SentenceTransformers @tomaarsen!